src/adapter/src/explain/optimizer_trace.rs

```diff
@@ -190,32 +181,13 @@ impl OptimizerTrace {
                 .map(|mut entry| {
                     // update the context with the current time
                     context.duration = entry.full_duration;
-                    match fast_path_plan {
-                        Some(fast_path_plan)
-                            if !context.config.no_fast_path && {
-                                [
-                                    ExplainStage::OptimizedPlan.path().expect("path"),
-                                    ExplainStage::PhysicalPlan.path().expect("path"),
-                                ]
-                                .contains(&entry.path.as_str())
-                            } =>
-                        {
-                            Ok(TraceEntry {
-                                instant: entry.instant,
-                                span_duration: entry.span_duration,
-                                full_duration: entry.full_duration,
-                                path: entry.path,
-                                plan: fast_path_plan.clone(),
-                            })
-                        }
-                        _ => Ok(TraceEntry {
-                            instant: entry.instant,
-                            span_duration: entry.span_duration,
-                            full_duration: entry.full_duration,
-                            path: entry.path,
-                            plan: Explainable::new(&mut entry.plan).explain(format, context)?,
-                        }),
-                    }
+                    Ok(TraceEntry {
```

Comment:
This is simplified because the removed logic now lives above:

https://github.com/MaterializeInc/materialize/pull/24887/files#diff-9a169fe3ef85a53dbf876b6b69d8d4f6f6c78901afcdf2683f72066e924b9df2R926-R932